app/components/Form/AmountInput.js

```diff
@@ -0,0 +1,134 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import CurrencyIcon from 'components/CurrencyIcon'
+import styles from './RequestForm.scss'
+
+class AmountInput extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this)
+    this.handleKeyDown = this.handleKeyDown.bind(this)
+  }
+
+  handleChange(e) {
+    let value = e.target.value;
+    let integer = null;
+    let fractional = null;
+
+    if (value.match(/[0-9]+/)) {
+      integer = e.target.value * 1;
+    }
+    if (value.match(/[0-9]*[,\.][0-9]+/)) {
+      [integer, fractional] = value.toString().split('.');
+    }
+
+    // limit fractional precision to 8 places
+    if (fractional && fractional.length > 8) {
```

Comment:
Limit is 11 characters for off-chain, 8 for on-chain.
https://github.com/lightningnetwork/lnd/blob/master/lnwire/msat.go

Reply:
This is incorrect - satoshis are still the unit for off-chain payments. Higher precision is used to accommodate fees.